src/gpu/intel/jit/ir/include/type.hpp

```diff
@@ -0,0 +1,436 @@
+/*******************************************************************************
+* Copyright 2021-2025 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_INTEL_JIT_IR_INCLUDE_TYPE_HPP
+#define GPU_INTEL_JIT_IR_INCLUDE_TYPE_HPP
+
+#include <cstdint>
+#include <string>
+
+#include "gpu/intel/jit/utils/utils.hpp"
+
+namespace ngen {
+enum class DataType : uint8_t;
+}
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace intel {
+namespace jit {
+
+namespace type {
+enum class attr_t : uint32_t { undef = 0, ptr = 1, mut = 2, simd = 4, slm = 8 };
```

Comment:
Nit: Looks like now all attribute usages are via `type::attr_t` vs `type_attr_t` previously, it's virtually the same but just one letter longer to type.